src/context.cpp

```diff
@@ -91,6 +91,12 @@ namespace oxen::quic
         stream_data_cb = std::move(func);
     }
 
+    void ServerContext::handle_serverctx_opt(stream_open_callback_t& func)
```

Comment:
This should be taken by value because, if it's a non-const lvalue reference, then you could end up stealing my value away from me.  For instance if I did this:
```C++
std::function<uint64_t(Stream&)> open_callback = [](Stream&) { /*... */; };

std::list<Server> servers;
for (int i = 0; i < 3; i++) {
    servers.emplace(open_callback);
}
```
then the *first* Server construction would end up stealing my function and leaving my `open_callback` empty, so that an empty function gets passed into the next ones.  (The same goes for the other callbacks, though I changed those in #1).

Reply:
> (The same goes for the other callbacks, though I changed those in https://github.com/oxen-io/oxen-libquic/pull/1)

Exactly why I did it the same (less optimal) way! They're all overloads of the same function, so I figured it would be easier to change all the signatures in #1